### PR TITLE
Fix for Get-SddcDiagnosticInfo to work in ConstrainedLanguage

### PR DESCRIPTION
**Pull request by kagoyal**
1. Pass ConfigurationName to all commands execution.
2. Pass CommonFunc as InitBlock only as ([scriptblock]::Create($using:CommonFunc)) is not supported in constrained language.
3. Use $env:temp instead of GetTempPath as method invocation is not supported.

### CODE REVIEW

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 1]
```diff
@@ -39,8 +39,6 @@ $CommonFuncBlock = {
     Import-Module SmbWitness
```
Comment (visvel) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1:39]: You might need to bump up the version in psd1 file for any new changes

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 2]
```diff
@@ -660,8 +658,15 @@ function Check-ExtractZip(
 
         Show-Update "Extracting $Path -> $ExtractToPath"
 
-        try { [System.IO.Compression.ZipFile]::ExtractToDirectory($Path, $ExtractToPath) }
-        catch { Show-Error("Can't extract results as Zip file from '$Path' to '$ExtractToPath'") }
+        try 
+        { 
+            Add-Type -Assembly System.IO.Compression.FileSystem
```
Comment (visvel) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1:663]: We might just need to do this Add-Type just once instead of in all places?

Comment (kagoyal) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1:663]: After removing script block instantiation, probably we don't need this everywhere.
Sure, let me try that.

Comment (kagoyal) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1:663]: It cannot be placed in common block, as initblock is executed in untrusted execution.
I have called add-type everywhere before using compression command, anything specific you have in mind?

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psd1 [hunk 3]
```diff
@@ -10,7 +10,7 @@
 RootModule = 'PrivateCloud.DiagnosticInfo.psm1'
 
 # Version number of this module.
-ModuleVersion = '1.1.26'
+ModuleVersion = '1.1.27'
```
Comment (visvel) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psd1:13]: Please update this with latest signed bits. (have updated the version to 1.1.28)

**Pull request merged**